pyte/screens.py

```diff
@@ -1240,13 +1241,16 @@ def prev_page(self):
             mid = min(len(self.history.top),
                       int(math.ceil(self.lines * self.history.ratio)))
 
-            self.history.bottom.extendleft(reversed(self.buffer[-mid:]))
+            extension = [self.buffer[line] for line in reversed(range(self.lines - mid, self.lines))]
+            self.history.bottom.extendleft(extension)
             self.history = self.history \
                 ._replace(position=self.history.position - self.lines)
 
-            self.buffer[:] = list(reversed([
-                self.history.top.pop() for _ in range(mid)
-            ])) + self.buffer[:-mid]
+            for line in reversed(range(self.lines)):
```

Comment:
This is probably best described with two for statements. One for the line >= mid case, and the other for the line < mid.